### PR TITLE
Upgrade snowflake-sqlalchemy package

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
Due to [changes](https://github.com/snowflakedb/snowflake-sqlalchemy/issues/350) in `sqlalchemy==1.4.42` our CI is failing, since snowflake-sqlalchemy's rfc imports are failing and we cannot change the version of snowflake-sqlalchemy to anything other than 1.2.4 because airflow 2.2.5 has a constraint.
## What is the new behavior?
Upgraded the package to 1.4.3 where the fix is added.

## Does this introduce a breaking change?
Nope

### CODE REVIEW

File: python-sdk/pyproject.toml [hunk 1]
```diff
@@ -59,7 +59,7 @@ google = [
 ]
 snowflake = [
     "apache-airflow-providers-snowflake",
-    "snowflake-sqlalchemy>=1.2.0,<=1.2.4",
+    "snowflake-sqlalchemy>=1.4.3,<1.5.0",
```
Comment (pankajastro) [inline python-sdk/pyproject.toml:62]: does it cause any problems if we increase the version from 1.2.0 to 1.4.3 and a customer has some other things  which depend on 1.2.0 in the env?

File: python-sdk/pyproject.toml [hunk 2]
```diff
@@ -23,7 +23,7 @@ dependencies = [
     "pyarrow",
     "python-frontmatter",
     "smart-open",
-    "SQLAlchemy>=1.3.18"
+    "SQLAlchemy>=1.3.18,<1.4.42"
```
Comment (pankajastro) [inline python-sdk/pyproject.toml:26]: why we are restricting maybe a comment would be helpful

Comment (utkarsharma2) [reply python-sdk/pyproject.toml:26]: We have it as part of the PR description, but no harm in adding in the code as well, will do.

**Review by pankajastro — APPROVED** [review]

**Review by feluelle — COMMENTED** [review]
Should we have a ticket to unpin this? Do we need the `rfc` imports?

**Comment by utkarsharma2** [comment]
@feluelle 
1. We should add the ticket.
2. So we don't directly use `rfc` imports, `snowflake-sqlalchemy` uses it and we cannot change the version of `snowflake-sqlalchemy` to anything other than 1.2.4 because airflow 2.2.5 has a constraint.

**Comment by feluelle** [comment]
Thanks for the explanation. Could you add the 2nd statement to the PR description, please?

**Comment by utkarsharma2** [comment]
@feluelle Added it in the PR and comment

**Pull request merged**